src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -96,8 +98,11 @@ class TaskKiller @Inject() (
       toKill = appTasks
     )
 
-    appTasks.keys.find(id => !instanceTracker.hasSpecInstancesSync(id))
-      .map(id => Future.failed(PathNotFoundException(id)))
+    Future.traverse(appTasks.keys){ id =>
```

Comment:
Can you convert to async/await for all future based code? e.g. you should rarely ever use `Future.traverse` or any monad method on a future.